### PR TITLE
142 dzelge place select empty

### LINKED ISSUE #142: Places not loading on PlaceGroupChange
**Describe the bug**

The place select stays empty when selecting place groups are selected.

**To Reproduce**
1. Go to cyanoalert
2. Click on select places and select a group
3. See places select

**Expected behavior**
Places should show up

### PR DESCRIPTION
**Pull request by dzelge**
Fixes #142 : When accepting this PR the place select will be filled when place groups are selected (EDIT) in the viewer.

### CODE REVIEW

File: xcube/webapi/context.py [hunk 1]
```diff
@@ -293,7 +292,7 @@ def get_dataset_place_groups(self, ds_id: str, load_features=False) -> List[Dict
         return place_groups
 
     def get_dataset_place_group(self, ds_id: str, place_group_id: str, load_features=False) -> Dict:
-        place_groups = self.get_dataset_place_groups(ds_id, load_features=False)
+        place_groups = self.get_dataset_place_groups(ds_id, load_features=load_features)
```
Comment (forman) [inline xcube/webapi/context.py]: `load_features=False` was *not* intended, see following `for` loop, where `load_features` is used to only load the desired place.

File: xcube/webapi/controllers/catalogue.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def get_dataset_place_groups(ctx: ServiceContext, ds_id: str) -> List[GeoJsonFea
 
 
 def get_dataset_place_group(ctx: ServiceContext, ds_id: str, place_group_id: str) -> GeoJsonFeatureCollection:
-    place_group = ctx.get_dataset_place_group(ds_id, place_group_id, load_features=False)
+    place_group = ctx.get_dataset_place_group(ds_id, place_group_id, load_features=True)
```
Comment (forman) [inline xcube/webapi/controllers/catalogue.py]: Not intended! Avoids lazy loading. The problem must be solved differently.

File: xcube/webapi/res/openapi.yml [hunk 3]
```diff
@@ -370,6 +370,30 @@ paths:
         '404':
           description: Resource not found.
 
+  '/datasets/{dataset}/filter/{geometry}/stats/{type}':
```
Comment (forman) [inline xcube/webapi/res/openapi.yml]: What does this have to do with #142? 
Where does the requirement come from?

File: xcube/webapi/res/openapi.yml [hunk 4]
```diff
@@ -626,6 +650,20 @@ components:
...
+      required: true
+      schema:
+        type: string
+    geometry:
+      name: geometry
+      in: path
+      description: Geometry for Filtering a Dataset
+      required: true
+      schema:
+        type: string
```
Comment (forman) [inline xcube/webapi/res/openapi.yml]: What does this have to do with #142? 
Where does the requirement come from?

**Comment by dzelge** [comment]
Agree on the openapi. Will take that out. Without the change the features will not be filled and therefore the place select will stay empty on places change. 

 

 

Von: Norman Fomferra <notifications@github.com> 
Gesendet: Donnerstag, 8. August 2019 12:11
An: dcs4cop/xcube <xcube@noreply.github.com>
Cc: Helge Dzierzon <helge.dzierzon@brockmann-consult.de>; Author <author@noreply.github.com>
Betreff: Re: [dcs4cop/xcube] 142 dzelge place select empty (#143)

 

@forman commented on this pull request.

This doesn't look right.
Also, mixed in openapi.yml changes that were not dicussed, and not agreed.

  _____  

In xcube/webapi/context.py <https://github.com/dcs4cop/xcube/pull/143#discussion_r311956585> :

> @@ -293,7 +292,7 @@ def get_dataset_place_groups(self, ds_id: str, load_features=False) -> List[Dict
         return place_groups
 
     def get_dataset_place_group(self, ds_id: str, place_group_id: str, load_features=False) -> Dict:
-        place_groups = self.get_dataset_place_groups(ds_id, load_features=False)
+        place_groups = self.get_dataset_place_groups(ds_id, load_features=load_features)

load_features=False was not intended, see following for loop, where load_features is used to only load the desired place.

  _____  

In xcube/webapi/controllers/catalogue.py <https://github.com/dcs4cop/xcube/pull/143#discussion_r311957064> :

> @@ -124,7 +124,7 @@ def get_dataset_place_groups(ctx: ServiceContext, ds_id: str) -> List[GeoJsonFea
 
 
 def get_dataset_place_group(ctx: ServiceContext, ds_id: str, place_group_id: str) -> GeoJsonFeatureCollection:
-    place_group = ctx.get_dataset_place_group(ds_id, place_group_id, load_features=False)
+    place_group = ctx.get_dataset_place_group(ds_id, place_group_id, load_features=True)

Not intended! Avoids lazy loading. The problem must be solved differently.

  _____  

In xcube/webapi/res/openapi.yml <https://github.com/dcs4cop/xcube/pull/143#discussion_r311957435> :

> @@ -370,6 +370,30 @@ paths:
         '404':
           description: Resource not found.
 
+  '/datasets/{dataset}/filter/{geometry}/stats/{type}':

What does this have to do with #142 <https://github.com/dcs4cop/xcube/issues/142> ?
Where does the requirement come from?

  _____  

In xcube/webapi/res/openapi.yml <https://github.com/dcs4cop/xcube/pull/143#discussion_r311957504> :

> @@ -626,6 +650,20 @@ components:
       required: true
       schema:
         type: string
+    statstype:
+      name: type
+      in: path
+      description: Type of Summary Statistics
+      required: true
+      schema:
+        type: string
+    geometry:
+      name: geometry
+      in: path
+      description: Geometry for Filtering a Dataset
+      required: true
+      schema:
+        type: string

What does this have to do with #142 <https://github.com/dcs4cop/xcube/issues/142> ?
Where does the requirement come from?

File: xcube/webapi/res/openapi.yml (hunk 4 above)
Comment (dzelge) [reply xcube/webapi/res/openapi.yml]: These changes were ideas related to dev work related to the [Xcube viewer issue 50](https://github.com/dcs4cop/xcube-viewer/issues/50) based on discussions with kerstin. No active development.

**Comment by dzelge** [comment]
Replaced by #145
